pkg/controller/controller.go

```diff
@@ -112,16 +118,25 @@ func (haproxy *HAProxyController) ConfigureFlags(flags *pflag.FlagSet) {
 
 // OverrideFlags allows controller to override command line parameter flags
 func (haproxy *HAProxyController) OverrideFlags(flags *pflag.FlagSet) {
-	if *haproxy.reloadStrategy == "native" || *haproxy.reloadStrategy == "reusesocket" {
-		haproxy.configFile = "/etc/haproxy/haproxy.cfg"
-		haproxy.template = newTemplate("haproxy.tmpl", "/etc/haproxy/template/haproxy.tmpl")
-	} else if *haproxy.reloadStrategy == "multibinder" {
-		haproxy.configFile = "/etc/haproxy/haproxy.cfg.erb"
-		haproxy.template = newTemplate("haproxy.cfg.erb.tmpl", "/etc/haproxy/haproxy.cfg.erb.tmpl")
-	} else {
+	haproxy.configDir = "/etc/haproxy"
+	haproxy.configFilePrefix = "haproxy"
+	haproxy.configFileSuffix = ".cfg"
+	haproxy.template = newTemplate("haproxy.tmpl", "/etc/haproxy/template/haproxy.tmpl")
+
+	if !(*haproxy.reloadStrategy == "native" || *haproxy.reloadStrategy == "reusesocket" || *haproxy.reloadStrategy == "multibinder") {
 		glog.Fatalf("Unsupported reload strategy: %v", *haproxy.reloadStrategy)
 	}
+	
+	if *haproxy.reloadStrategy == "multibinder" {
+		haproxy.template = newTemplate("haproxy.cfg.erb.tmpl", "/etc/haproxy/haproxy.cfg.erb.tmpl")
+	}
 	haproxy.command = "/haproxy-reload.sh"
+
+	haproxy.maxOldConfigFiles = 1000
+	envVar := os.Getenv("MAX_OLD_CONFIG_FILES")
+	if maxOldConfigFiles, err := strconv.Atoi(envVar); err == nil {
+		haproxy.maxOldConfigFiles = maxOldConfigFiles
+	}
```

Comment:
use a command-line argument instead, declare at ConfigureFlags func just below reloadStrategy.

I'd also like to see a zero option, which means no old backup config should be used. I have some manual testing scenarios where I need a known filename, perhaps others could have similar issues.